cmake/tpl/FindSUPERLUDIST.cmake

```diff
@@ -14,94 +14,103 @@
-endif()
-
-# set the include dir option if it wasn't preset
-if(SUPERLUDIST_LIBRARY AND (NOT SUPERLUDIST_INCLUDE_DIR))
-  get_filename_component(SUPERLUDIST_INCLUDE_DIR ${SUPERLUDIST_LIBRARY_DIR} DIRECTORY)
-  set(SUPERLUDIST_INCLUDE_DIR "${SUPERLUDIST_INCLUDE_DIR}/include" CACHE PATH "" FORCE)
+if(NOT SUPERLUDIST_LINK_LIBRARIES AND SUPERLUDIST_LIBRARIES)
+  set(SUPERLUDIST_LINK_LIBRARIES "${SUPERLUDIST_LIBRARIES}" CACHE INTERNAL "")
+elseif(NOT SUPERLUDIST_LINK_LIBRARIES)
+  find_package(PkgConfig REQUIRED)
+  list(APPEND CMAKE_PREFIX_PATH "${SUPERLUDIST_DIR}")
+  pkg_search_module(SUPERLUDIST REQUIRED "superlu_dist >= 7.0.0")
+  set(SUPERLUDIST_LINK_LIBRARIES "${SUPERLUDIST_LINK_LIBRARIES}" CACHE INTERNAL "")
+  set(SUPERLUDIST_INCLUDE_DIRS "${SUPERLUDIST_INCLUDE_DIRS}" CACHE INTERNAL "")
 endif()
 
 # find the library configuration file
-if(SUPERLUDIST_LIBRARY AND SUPERLUDIST_INCLUDE_DIR)
-  find_file(SUPERLUDIST_CONFIG_PATH superlu_dist_config.h PATHS ${SUPERLUDIST_INCLUDE_DIR})
-  file(STRINGS ${SUPERLUDIST_CONFIG_PATH} _strings_with_index_size REGEX "XSDK_INDEX_SIZE")
-  list(GET _strings_with_index_size 0 _index_size_string)
-  string(REGEX MATCHALL "[0-9][0-9]" SUPERLUDIST_INDEX_SIZE "${_index_size_string}")
+set(SUPERLUDIST_CUDA FALSE CACHE INTERNAL "SuperLU DIST was built with CUDA support")
+set(SUPERLUDIST_ROCM FALSE CACHE INTERNAL "SuperLU DIST was built with ROCm support")
+if(SUPERLUDIST_INCLUDE_DIRS)
+  find_file(SUPERLUDIST_CONFIG_PATH superlu_dist_config.h PATHS "${SUPERLUDIST_INCLUDE_DIRS}")
   mark_as_advanced(FORCE SUPERLUDIST_CONFIG_PATH)
+  if(SUPERLUDIST_VERSION VERSION_GREATER_EQUAL "8.0.0")
+      file(STRINGS "${SUPERLUDIST_CONFIG_PATH}" _index_size_64 REGEX "#define XSDK_INDEX_SIZE 64")
+      file(STRINGS "${SUPERLUDIST_CONFIG_PATH}" _index_size_32 REGEX "#undef XSDK_INDEX_SIZE")
+      if(_index_size_64)
+        set(SUPERLUDIST_INDEX_SIZE 64 CACHE STRING "SuperLU DIST index size (bit width)" FORCE)
+      else()
+        set(SUPERLUDIST_INDEX_SIZE 32 CACHE STRING "SuperLU DIST index size (bit width)" FORCE)
+      endif()
+      mark_as_advanced(FORCE SUPERLUDIST_INDEX_SIZE)
+  else()
+    file(STRINGS "${SUPERLUDIST_CONFIG_PATH}" _strings_with_index_size REGEX "XSDK_INDEX_SIZE")
+    list(GET _strings_with_index_size 0 _index_size_string)
+    string(REGEX MATCHALL "[0-9][0-9]" SUPERLUDIST_INDEX_SIZE "${_index_size_string}")
+  endif()
+  file(STRINGS "${SUPERLUDIST_CONFIG_PATH}" _strings_have_cuda REGEX "HAVE_CUDA")
+  string(REGEX MATCH "TRUE|FALSE" _has_cuda "${_strings_have_cuda}")
+  file(STRINGS "${SUPERLUDIST_CONFIG_PATH}" _strings_have_rocm REGEX "HAVE_ROCM")
+  string(REGEX MATCH "TRUE|FALSE" _has_rocm "${_strings_have_rocm}")
+  set(SUPERLUDIST_CUDA ${_has_cuda} CACHE INTERNAL "SuperLU DIST was built with CUDA support" FORCE)
+  set(SUPERLUDIST_ROCM ${_has_rocm} CACHE INTERNAL "SuperLU DIST was built with ROCm support" FORCE)
+  unset(_has_cuda)
+  unset(_has_rocm)
```

Comment:
Is there a similar `HAVE` define for OpenMP?

Reply:
Unfortunately, no.